apiserver/params/cloud.go

```diff
@@ -114,3 +114,67 @@ type CloudSpecResult struct {
 type CloudSpecResults struct {
 	Results []CloudSpecResult `json:"results,omitempty"`
 }
+
+// CloudCredentialRequestParam defines a credential in terms of its cloud and name.
+// It is used to request detailed content for the credential stored on the controller.
+type CloudCredentialRequestParam struct {
+	CloudName      string `json:"cloud-name"`
+	CredentialName string `json:"credential-name"`
+}
+
+func (p CloudCredentialRequestParam) IsEmpty() bool {
+	return p.CloudName == "" && p.CredentialName == ""
+}
+
+// CredentialContentRequestParam defines an input required to make a valid call
+// to get credentials content stored on the controller.
+type CredentialContentRequestParam struct {
```

Comment:
We may not be consistent everywhere, but we have been tending to name this struct FooArgs and the slice as
args []FooArg
ymmv but would be nice to try and stay consistent

Reply:
I have seen both Args and Request-related naming. I'll change to Args but really does it matter if the naming is descriptive?..